modules/renter/contractor/allowance.go

```diff
@@ -147,12 +148,10 @@ func (c *Contractor) SetAllowance(a modules.Allowance) error {
 	c.mu.Lock()
 	c.allowance = a
 	c.contracts = newContracts
-	// update metrics
-	var spending types.Currency
-	for _, contract := range c.contracts {
-		spending = spending.Add(contract.RenterFunds())
+	// if the currentPeriod was previously unset, set it now
+	if c.currentPeriod == 0 {
+		c.currentPeriod = periodStart
```

Comment:
not sure. Maybe it should only reset if you replace all your contracts. The allowance has a lot of edge cases.